### DISCUSSION TITLE
Change Revenue/Expense to Asset account

### DISCUSSION THREAD

**Discussion by baflo** [firefly-iii/firefly-iii#7983] [General]
I had the situation that I was importing for months from one account and now wanted to add another account, too. However, this new account was already automatically added a expense account. So I wondered if I could change the account type.

The only information I found is from this issue, which turned out to be very manual work: https://github.com/firefly-iii/firefly-iii/issues/1700

Eventually I found another solution I want to show here. The fundamental problem is, that I cannot create a new asset account with the same IBAN. However, I _can_ create a new asset account without an IBAN, then change the destination account of all transactions to this one and eventually delete the automatically created account. This can be automated:

1. Create a new asset account without IBAN
2. Search for `destination_account_nr_is:"AX2112345600000785"`
3. Ensure the query finds the correct transactions, then click "Create new rule from search query"
4. Add rule to change destination account to the new created one (you may need to use the action "Convert the transactinon to a transfer")
5. Go to Automations -> Rules and select the "Standby" symbol to run the rule on all transactions matching the rule.
6. Go to Accounts -> expense/revenue accounts -> select old account and verify there are no transactions left
7. Delete old account
8. Add IBAN to new account

**Comment by JC5** [comment]
That's a very cool solution, thanks! I'll add this to the FAQ as well :+1:
